### ISSUE TITLE
TypeError: expected str, bytes or os.PathLike object, not NoneType

### ISSUE THREAD

**Issue by mr-hespicable** [daspartho/SpotiByeAds#26]
Whenever I attempt to run the code I get this error: `TypeError: expected str, bytes or os.PathLike object, not NoneType`.  
I run it as follows:  

$ cd C:\Users\me\SpotiByeAds

$ py main.py
Found previously used credentials. Want to use them again? (Y/n) Y

Awesome, that's all I needed. I'm watching for ads now <.<
SUCCESS: The process "Spotify.exe" with PID 12680 has been terminated.
SUCCESS: The process "Spotify.exe" with PID 23276 has been terminated.
SUCCESS: The process "Spotify.exe" with PID 25124 has been terminated.
SUCCESS: The process "Spotify.exe" with PID 11972 has been terminated.
SUCCESS: The process "Spotify.exe" with PID 20088 has been terminated.
SUCCESS: The process "Spotify.exe" with PID 13544 has been terminated.
Traceback (most recent call last):
  File "C:\Users\me\SpotiByeAds\main.py", line 128, in <module>
    main(spotify_username, spotifyAccessScope, spotify_client_id, spotify_client_secret, spotifyRedirectURI, PATH)
  File "C:\Users\me\SpotiByeAds\main.py", line 65, in main
    restartSpotify(path)
  File "C:\Users\me\SpotiByeAds\main.py", line 52, in restartSpotify
    openSpotify(path)
  File "C:\Users\me\SpotiByeAds\main.py", line 37, in openSpotify
    subprocess.Popen([path], start_new_session=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, shell=True)
  File "C:\Users\me\AppData\Local\Programs\Python\Python39\lib\subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:\Users\me\AppData\Local\Programs\Python\Python39\lib\subprocess.py", line 1360, in _execute_child
    args = list2cmdline(args)
  File "C:\Users\me\AppData\Local\Programs\Python\Python39\lib\subprocess.py", line 565, in list2cmdline
    for arg in map(os.fsdecode, seq):
  File "C:\Users\me\AppData\Local\Programs\Python\Python39\lib\os.py", line 822, in fsdecode
    filename = fspath(filename)  # Does type-checking of `filename`.
TypeError: expected str, bytes or os.PathLike object, not NoneType`

**Comment by lykkjur** [comment]
How do you have Spotify installed? Also, is Python on your `PATH`?

**Comment by Cyclip** [comment]
I have an error too, both are because the constant `path` remains as `None`:
```Traceback (most recent call last):
  File "main.py", line 146, in <module>
    main(spotify_username, SPOTIFY_ACCESS_SCOPE, spotify_client_id, spotify_client_secret, SPOTIFY_REDIRECT_URI, PATH)
  File "main.py", line 89, in main
    restartSpotify(path)
  File "main.py", line 73, in restartSpotify
    openSpotify(path)
  File "main.py", line 52, in openSpotify
    subprocess.Popen([path], start_new_session=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, shell=True)
  File "C:\Users\Cyclip\AppData\Local\Programs\Python\Python37\lib\subprocess.py", line 800, in __init__
    restore_signals, start_new_session)
  File "C:\Users\Cyclip\AppData\Local\Programs\Python\Python37\lib\subprocess.py", line 1148, in _execute_child
    args = list2cmdline(args)
  File "C:\Users\Cyclip\AppData\Local\Programs\Python\Python37\lib\subprocess.py", line 555, in list2cmdline
    needquote = (" " in arg) or ("\t" in arg) or not arg
TypeError: argument of type 'NoneType' is not iterable```

**Comment by lykkjur** [comment]
Oh I'm dumb

**Comment by AnonymouX47** [comment]
> both are because the constant path remains as None.
It's caused by main.py: line 33, when `shutil.which()` can't find the command.
Fixed it in a #29 commit https://github.com/daspartho/SpotiByeAds/pull/29/commits/77c0e3dd983e91424c2865e7d363b4d3c84cc03e but not yet yet merged.
Also, i think #31 might've fixed it on windows, according to the author of the PR.

**Comment by EtrieusVet** [comment]
Is there a fix yet?

**Comment by AnonymouX47** [comment]
Well... i guess i fixed this in https://github.com/daspartho/SpotiByeAds/pull/29/commits/0b5668ce05c2c21b61aea7c92b179542be6649cf, as long as you didn't install from Windows Store (I don't know the installation path for this case).

The PR is yet to be merged, For the mean time, you can probably clone my Fork (and switch to branch `patch-1`).

**Comment by jedistack** [comment]
my fix for the `shutil.which()` problem was kind pretty silly, but since the function is just supposed to find the path and can't, you just just find it yourself and set `path = C:/Users/[username]/AppData/Roaming/Spotify/Spotify.exe`
There are probably more elegant solutions, but it's working so far

**Comment by AnonymouX47** [comment]
Oh, sorry.... it's been fixed (#29).

Either do a `git pull` or clone the repo afresh (i.e follow the installation instructions again).
